### PR TITLE
Fix failed utils tests on WIN & QT platform

### PR DESCRIPTION
**Pull request by OHerasym**
### PR Changes:
- Fix failed utils tests on WIN & QT platform

Related: [APPLINK-27145](https://adc.luxoft.com/jira/browse/APPLINK-27145), [27156](https://adc.luxoft.com/jira/browse/APPLINK-27156)

### CODE REVIEW

**Comment by OHerasym** [comment]
Please review: @anosach-luxoft , @LuxoftAKutsan , @orhan-mehmedov, @dtrunov

File: src/components/utils/test/date_time_test.cc [hunk 1]
```diff
@@ -108,11 +108,10 @@ TEST(DateTimeTest, CalculateTimeSpan) {
   // arrange
   const TimevalStruct time = date_time::DateTime::getCurrentTime();
 
-  const uint32_t sleep_time_mSec = 10;
+  const uint64_t sleep_time_mSec = 100;
```
Comment (orhan-mehmedov) [inline src/components/utils/test/date_time_test.cc]: @OHerasym Why? Is adding more time just solves some unit test problems (for now)?

File: src/components/utils/test/date_time_test.cc [hunk 2]
```diff
@@ -108,11 +108,10 @@ TEST(DateTimeTest, CalculateTimeSpan) {
...
   threads::sleep(sleep_time_mSec);
-
   // assert
-  ASSERT_GE(date_time::DateTime::calculateTimeSpan(time), sleep_time_mSec);
+  ASSERT_GE(date_time::DateTime::calculateTimeSpan(time) + 1, sleep_time_mSec);
```
Comment (orhan-mehmedov) [inline src/components/utils/test/date_time_test.cc]: @OHerasym Why +1? Can you clarify is there requirement / design about calculateTimeSpan() accuracy?

Comment (pestOO) [inline src/components/utils/test/date_time_test.cc]: @orhan-mehmedov @OHerasym 
Looks like we need just `ASSERT_GT` isntead of `ASSERT_GE`.

**Comment by dtrunov** [comment]
@OHerasym Reviewed

File: src/components/utils/test/date_time_test.cc (hunk 2 above)
Comment (OHerasym) [inline src/components/utils/test/date_time_test.cc]: @orhan-mehmedov , @pestOO , The problem is how we get current time, on windows platform we call GetSystemTimeAsFileTime(). This method returns time in nanoseconds. When we convert nanoseconds to miliseconds, we just div it to 1000. 

For example: we call sleep(30) // wait 30 miliseconds
GetSystemTimeAsFileTime() - can return 29.9999997 / 1000 = 29

and test will fail.

File: src/components/utils/test/date_time_test.cc (hunk 1 above)
Comment (OHerasym) [inline src/components/utils/test/date_time_test.cc]: @orhan-mehmedov , on QT platform, sleep(10) , sometimes works inncorect. thread can sleep only 1 ms.  I read that sleep() with values < 15 , can work wrong.

Comment (orhan-mehmedov) [inline src/components/utils/test/date_time_test.cc]: @OHerasym Well, if this is the case my advice is - do not dig in such know-how silently in the code. Please, add at least comment. My other advice is if you found this - what about ticket/proposal SDL to be reviewed is there sleep below 15ms and to be discussed what can be done etc. But at least add a comment here why 100 was choosen.

File: src/components/utils/test/stl_utils_test.cc [hunk 3]
```diff
@@ -42,17 +42,17 @@ namespace utils_test {
 using ::utils::StlCollectionDeleter;
 using ::utils::StlMapDeleter;
 
-class TestObject {
+class TestSTLObject {
  public:
-  ~TestObject() {}
+  ~TestSTLObject() {}
 };
```
Comment (anosach-luxoft) [inline src/components/utils/test/stl_utils_test.cc]: @OHerasym please put it to anonymous namespace

Comment (OHerasym) [inline src/components/utils/test/stl_utils_test.cc]: @anosach-luxoft , 96e2eb7, Done

File: src/components/include/utils/threads/message_loop_thread.h [hunk 4]
```diff
@@ -134,9 +134,15 @@ size_t MessageLoopThread<Q>::GetMessageQueueSize() const {
 template <class Q>
 MessageLoopThread<Q>::MessageLoopThread(const std::string& name,
                                         Handler* handler,
-                                        const ThreadOptions& thread_opts)
-    : thread_delegate_(new LoopThreadDelegate(&message_queue_, handler))
-    , thread_(threads::CreateThread(name.c_str(), thread_delegate_)) {
+                                        const ThreadOptions& thread_opts,
+                                        void* loop_delegate) {
```
Comment (LuxoftAKutsan) [inline src/components/include/utils/threads/message_loop_thread.h]: @OHerasym do we need custom loop_delegate_for tests?

File: src/components/utils/test/date_time_test.cc (hunk 1 above)
Comment (LuxoftAKutsan) [inline src/components/utils/test/date_time_test.cc]: @OHerasym I also don't like idea ti increase timeout due to unknown reason of QT behaviour

File: src/components/utils/test/date_time_test.cc (hunk 2 above)
Comment (LuxoftAKutsan) [inline src/components/utils/test/date_time_test.cc]: @OHerasym Please avoid magic numbers. Try to use may be some variable with description why do you need them

File: src/components/utils/test/include/utils/test_handler.h [hunk 5]
```diff
@@ -40,9 +40,25 @@ namespace components {
 namespace utils_test {
 
 typedef std::queue<bool> BoolQueue;
-typedef threads::MessageLoopThread<BoolQueue> TestLoopThread;
+typedef threads::MessageLoopThread<BoolQueue> BoolLoopThread;
```
Comment (LuxoftAKutsan) [inline src/components/utils/test/include/utils/test_handler.h]: @OHerasym Bool loop thread? Werry unclear, At least need descriptions

File: src/components/utils/test/stl_utils_test.cc [hunk 6]
```diff
@@ -62,8 +62,8 @@ TEST(StlDeleter, DestructMapWithOneElement) {
 
 TEST(StlDeleter, DestructMapWithSeveralElements) {
   TestMap test_map;
-  test_map[1] = new TestObject();
-  test_map[2] = new TestObject();
+  test_map[1] = new TestSTLObject();
```
Comment (LuxoftAKutsan) [inline src/components/utils/test/stl_utils_test.cc]: @OHerasym Is it rely impossible to avoid new\delete?

**Comment by OHerasym** [comment]
Pull request is blocked by [this issue](https://adc.luxoft.com/jira/browse/APPLINK-28086)

File: src/components/utils/include/utils/convert_utils.h [hunk 7]
```diff
@@ -61,6 +61,11 @@ unsigned long long int ConvertUInt64ToLongLongUInt(const uint64_t value);
 uint64_t ConvertLongLongUIntToUInt64(const unsigned long long int value);
 
 /**
+ * Return integer division round up
```
Comment (pestOO) [inline src/components/utils/include/utils/convert_utils.h]: @OHerasym 
Please, provide more detail information about parameters.
Maybe one example of usage?

File: src/components/utils/src/convert_utils.cc [hunk 8]
```diff
@@ -70,3 +70,9 @@ uint64_t utils::ConvertLongLongUIntToUInt64(
 
   return static_cast<uint64_t>(value);
 }
+
+int64_t utils::Int64DivisionRoundUp(const int64_t numerator,
+                                    const int64_t denominator) {
```
Comment (pestOO) [inline src/components/utils/src/convert_utils.cc]: @OHerasym,
Maybe better to set `denominator` as unsigned?

File: src/components/utils/include/utils/convert_utils.h (hunk 7 above)
Comment (pestOO) [inline src/components/utils/include/utils/convert_utils.h]: Or better to add UTests with usage of `Int64DivisionRoundUp`

**Pull request closed without merging**